bleach/__init__.py

```diff
@@ -347,6 +347,13 @@ def link_repl(match):
         if url.startswith('('):
             _wrapping = strip_wrapping_parentheses(url)
             url, open_brackets, close_brackets = _wrapping
+        elif url.endswith('/)'):
+            # This is a very common corner case where the URL is listed
+            # last within a longer text within parentheses.
+            # Example: (check my site at http://foo.bar/)
+            # Do not consider the closing ) as part of the URL.
+            url, open_brackets, close_brackets = url[:-1], 0, 1
```

Comment:
This fix isn't related to the rest of this PR. I'm not sure I understand the issue here and I'm not wild about this fix either. If I do understand the issue correctly, seems like this is a problem with the regex and should be dealt with there.

I think it's probably worth opening an issue for this if there isn't one already so we can figure that out separately and removing this bit from the PR.